service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -282,13 +253,39 @@ private String prepareStatementsForDerivedObservations(IdMapper idMapper, List<O
                     )
                 );
             } else {
-                derivedObservationsBlock.append(observationStatement).append(narrativeStatement);
+                derivedObservationsBlock.append(observationStatement).append(narrativeStatements);
             }
         });
 
         return derivedObservationsBlock.toString();
     }
 
+    private boolean observationDoesNotHaveCommentNoteCode(Observation observation) {
+        return observation.hasCode() && !CodeableConceptMappingUtils.hasCode(observation.getCode(), List.of(COMMENT_NOTE_CODE));
+    }
```

Comment:
Wouldn't `!observation.hasCode()` also satisfy `observationDoesNotHaveCommentNoteCode`?
```suggestion
    private boolean observationDoesNotHaveCommentNoteCode(Observation observation) {
        return !observation.hasCode() || !CodeableConceptMappingUtils.hasCode(observation.getCode(), List.of(COMMENT_NOTE_CODE));
    }
```

Or are we looking for something like `observationHasNonCommentNoteCode`?
```suggestion
    private boolean observationHasNonCommentNoteCode(Observation observation) {
        return observation.hasCode() && !CodeableConceptMappingUtils.hasCode(observation.getCode(), List.of(COMMENT_NOTE_CODE));
    }
```

Reply:
I like `observationHasNonCommentNoteCode`, I'll change it to that